Subject: details: fix detection of firmware auto-selection
From: Pavel Hrdina phrdina@redhat.com Tue Jul 21 12:33:56 2020 +0200
Date: Thu Aug 13 16:40:53 2020 +0200:
Git: e8bf16b983558010cc9645e09eb36117e9e4fba4

Commit <15a9502b7b7a263c4d66ff2b3f31c209f58fe0b4> fixed firmware
detection but incorrectly. It will always show only "UEFI" even if
the firmware auto-selection is not used because the function is_uefi()
checks both the old style and the new auto-selection.

We have to check only for the auto-selection option.

Signed-off-by: Pavel Hrdina <phrdina@redhat.com>

--- virt-manager-2.2.1.orig/virtManager/details/details.py
+++ virt-manager-2.2.1/virtManager/details/details.py
@@ -2017,7 +2017,7 @@ class vmmDetails(vmmGObjectUI):
 
         # Firmware
         domcaps = self.vm.get_domain_capabilities()
-        if self.vm.get_xmlobj().is_uefi():
+        if self.vm.get_xmlobj().os.firmware == "efi":
             firmware = 'UEFI'
         else:
             firmware = domcaps.label_for_firmware_path(
